packages/plugin-essentials/sources/commands/unlink.ts

```diff
@@ -0,0 +1,78 @@
+import {BaseCommand, WorkspaceRequiredError}                      from '@yarnpkg/cli';
+import {Cache, Configuration, Project, StreamReport, structUtils} from '@yarnpkg/core';
+import {npath, ppath}                                             from '@yarnpkg/fslib';
+import {Command, Option, Usage, UsageError}                       from 'clipanion';
+
+// eslint-disable-next-line arca/no-default-export
+export default class UnlinkCommand extends BaseCommand {
+  static paths = [
+    [`unlink`],
+  ];
+
+  static usage: Usage = Command.Usage({
+    description: `disconnect the local project from another one`,
+    details: `
+      This command will remove any resolutions in the project-level manifest that would have been added via a yarn link with similar arguments.
+    `,
+    examples: [[
+      `Unregister a remote workspace in the current project`,
+      `$0 unlink ~/ts-loader`,
+    ], [
+      `Unregister all workspaces from a remote project in the current project`,
+      `$0 unlink ~/jest --all`,
+    ]],
+  });
+
+  all = Option.Boolean(`-A,--all`, false, {
+    description: `Unlink all workspaces belonging to the target project from the current one`,
+  });
+
+  destination = Option.String();
```

Comment:
What about making `destination` optional, in which case it removes all portals in the `resolutions` field?

Reply:
It would also be a good idea imo to support package names as well as paths (ie either of `yarn unlink jest` or `yarn unlink ~/jest`).

Reply:
I made destination optional. I was thinking about package name support originally, but I'm not sure how to match the package names against the "pattern" in the resolution field. Is there a function for this?